www/api/ogBlog.ts

```diff
@@ -0,0 +1,57 @@
+import { ImageResponse } from '@vercel/og';
+import { OGBlogComponent } from './ogComponents';
+
+export const config = {
+  runtime: 'experimental-edge',
+};
+
+const inter = fetch(new URL('../static/Inter.ttf', import.meta.url)).then(
+  (res) => res.arrayBuffer(),
+);
+
+const alex = 'https://avatars.githubusercontent.com/u/459267?v=4';
+
+export default async (req: Request) => {
+  const interData = await inter;
+
+  const { searchParams } = new URL(req.url);
+  console.log({ searchParams });
+
+  const title = searchParams.get('title') ?? 'Default tRPC Title';
+
+  const date = Intl.DateTimeFormat('en-US', {
+    month: 'long',
+    day: 'numeric',
+    year: 'numeric',
+  }).format(
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    searchParams.has('date') ? new Date(searchParams.get('date')!) : new Date(),
+  );
```

Comment:
We can use zod to parase the search params!

```suggestion
export const paramsSchema = z.object({
  title: z.string(),
  date: z.coerce.date(),
  // [...]
});

export default async (req: Request) => {
  const interData = await inter;

  const url = new URL(req.url);

  const params = paramsSchema.parse(Object.fromEntries(url.searchParams))
  
  const date = Intl.DateTimeFormat('en-US', {
    month: 'long',
    day: 'numeric',
    year: 'numeric',
  }).format(
    params.date,
  );
```

Reply:
Maybe `.safeParse` and throw a 400 with the errors too 🤷